src/battle.ts

```diff
@@ -3180,8 +3201,8 @@ class Battle {
 			break;
 		}
 		case 'join': case 'j': case 'J': {
-			if (this.roomid) {
-				let room = app!.rooms[this.roomid];
+			if (this.roomid && window.app) {
```

Comment:
`this.roomid` should not exist if `window.app` doesn't, at least that's the theory behind the `!`. But I guess the extra safety couldn't hurt.

Reply:
you have no idea how external users will set `roomid` - your logic only holds in the PS UI

Reply:
There's no reason to set `roomid` except to communicate back with a PS room. I might as well have no idea that people will override random functions with functions that throw, and defend against that or something. There's no way to be secure against enemies with write access to your own code; why bother?

Reply:
LOL 'enemies'? are you kidding? Of course there are valid reasons to use the roomid field. obviously your general point that anything could be overriden is valid, but if thats only logical if theyre breaking the contract laid out by the class. Right now, the fields contract is "this only has meaning in  PS" which i disagree with, as thats entirely against the point of what I'm trying to do here with this package.

> But I guess the extra safety couldn't hurt.

lets just leave this as the reason then

Reply:
> Of course there are valid reasons to use the roomid field.

Can you give an example of a valid reason to use the `roomid` field other than to interface with `app.rooms`?

> Right now, the fields contract is "this only has meaning in PS" which i disagree with

I don't understand how you can disagree with a field's contract? I went back and checked, and it literally does nothing except interface with `app.rooms`. Go make your own field if you want one with a different contract.

I don't understand how this is any different from typing `throw new Error()` and being mad when it throws an error.

> lets just leave [extra safety] as the reason then

Actually, no, that was before I took a second look at the code. I'm now actively confused. This isn't safety, it's insanity.

`battle.roomid` does nothing _except_ report joins/leaves back to `app.rooms[roomid]`. It could certainly be documented better, but whether or not it is, it seems insane to `console.log("Hello world")` and then complain when `"Hello world"` shows up in your console.

Throwing an error when you set `battle.roomid` without `app.rooms[roomid]` existing is what _should_ happen. You shouldn't silently suppress errors.